core/modules/image/image.tokens.inc

```diff
@@ -0,0 +1,324 @@
+                    $output[] = $field[$attribute];
+                  }
+              }
+            }
+            else {
+              $output[] = image_style_url($type, $field['uri']);
+            }
+          }
+
+          // Compile each of the elements into a comma separated list, per
+          // standard Token practice.
+          $replacements[$original] = implode(', ', $output);
+        }
+      }
+    }
+  }
+
+  return $replacements;
+}
+
+/**
+ * Helper function for getting all available image attributes.
+ *
+ * @return array
+ *   All of the available image attributes.
+ */
+function _image_style_token_image_attributes() {
+  return array(
+    'style_name' => NULL,
+    'uri' => '',
+    'path' => NULL,
+    'path:relative' => NULL,
+    'width' => NULL,
+    'height' => NULL,
+    'extension' => NULL,
+    'filesize' => NULL,
+    'alt' => '',
+    'title' => NULL,
+    'mimetype' => NULL,
+    'first' => '',
+    'second' => '',
+    'third' => '',
+    'last' => '',
+  );
+}
+
+/**
+ * Helper function to generate image style if does not exist.
+ *
+ * @param string $styled_uri
+ *   URI for image style.
+ * @param string $type
+ *   Machine name for image style.
+ * @param string $field_uri
+ *   URI of field to be styled.
+ */
+function _image_generate_image($styled_uri, $type, $field_uri) {
+  // Only generate the image if it doesn't already exist.
+  if (!file_exists($styled_uri)) {
+    $style = image_style_load($type);
+    // Create a derivative of the image for this style.
+    image_style_create_derivative($style, $field_uri, $styled_uri);
+  }
+}
+
+/**
+ * Helper function to find all image fields.
+ *
+ * @return array
+ *   An array containing all image fields. Keys of the array is an identifier
+ *   made up of field_type:field_name:entity_type:bundle. Values are the
+ *   human-readable name of the field.
+ */
+function _image_get_image_fields() {
+  $image_fields = array();
+
+  $fields = field_info_fields();
+  foreach ($fields as $field) {
+    foreach ($field['bundles'] as $entity_type => $bundles) {
+      foreach ($bundles as $bundle) {
+        $key = $field['type'] . ':' . $field['field_name'] . ':' . $entity_type . ':' . $bundle;
+        $label = $entity_type . ': ' . $bundle . ': ' . $field['field_name'];
+        if ($field['type'] == 'image') {
+          $image_fields[$key] = $label;
+        }
+      }
+    }
+  }
+
+  return $image_fields;
+}
+
+/**
+ * Implements hook_token_info_alter().
+ */
+function image_token_info_alter(&$data) {
+  $fields = field_info_field_map();
+  $entity_info = entity_get_info();
+
+  // Check to see which fields are supported.
+  $supported_fields = _image_get_image_fields();
+  $supported_fields = array_filter($supported_fields);
+
+  foreach ($fields as $field_name => $field) {
+    $field['field_name'] = $field_name;
+    if ($field['type'] == 'image' || $field['type'] == 'file' || $field['type'] == 'media') {
```

Comment:
I think `media` should be removed here and in subsequent lines. Contrib Media module would have to re-implement similar token support.

Reply:
media removed.